### PR TITLE
CODENVY-1253 Add ability to configure workspace idle timeout per account

### PR DESCRIPTION
**Pull request by sleshchenko**
### What does this PR do?
Add ability to configure workspace idle timeout per account. 
By default users' workspaces will have timeout that are configured by `limits.workspace.idle.timeout` configuration property (following codenvy environment property `CODENVY_LIMITS_WORKSPACE_IDLE_TIMEOUT` that is renamed from `CODENVY_MACHINE_WS_AGENT_INACTIVE_STOP_TIMEOUT_MS`).
User who has `manageSystem` permissions is able to override this value for an user by sending POST request to {HOST}/api/resources/free with following body
```
{
  "accountId": "{USER_ID_HERE}",
  "resources": [
    {
      "type": "timeout",
      "amount": {REQUIRED_IDLE_TIMEOUT_HERE},
      "unit": "minute"
    }
  ]
}
```
P.S. This PR also contains changes that fix small issues with resources validation and error messages when user doesn't have enough resources to start or create workspace.

### What issues does this PR fix or reference?
https://github.com/codenvy/codenvy/issues/1253

#### Changelog
Added ability to configure workspace idle timeout per account.

#### Release Notes
Added ability to configure workspace idle timeout per account.

#### Docs PR
https://github.com/codenvy/docs/pull/79

### CODE REVIEW

File: wsmaster/codenvy-hosted-api-organization/src/main/java/com/codenvy/organization/api/resource/DefaultOrganizationResourcesProvider.java [hunk 1]
```diff
@@ -42,16 +43,21 @@
  */
 @Singleton
 public class DefaultOrganizationResourcesProvider implements DefaultResourcesProvider {
+    private static final long MILLISECONDS_TO_SECONDS_DIVIDER = 60 * 1000;
```
Comment (skabashnyuk) [inline wsmaster/codenvy-hosted-api-organization/src/main/java/com/codenvy/organization/api/resource/DefaultOrganizationResourcesProvider.java]: TimeUnit.SECONDS.convert(elapsedTime, TimeUnit.NANOSECONDS)

**Comment by sleshchenko** [comment]
@TylerJewell Please review new property name

File: wsmaster/codenvy-hosted-api-organization/src/main/java/com/codenvy/organization/api/resource/SuborganizationResourcesProvider.java [hunk 2]
```diff
@@ -29,13 +31,20 @@
...
+import java.util.stream.Collectors;
 
 import static java.util.Collections.singletonList;
 
 /**
- * Provides resources that are shared for suborganization by its parent organization
+ * Provides resources that are distributed for suborganization by its parent organization.
+ *
+ * <p>By default suborganizations are not able to use parent's resources,
+ * except timeout resource. Suborganizations inherits timeout resource
```
Comment (evoevodin) [inline wsmaster/codenvy-hosted-api-organization/src/main/java/com/codenvy/organization/api/resource/SuborganizationResourcesProvider.java]: typo _interit_

File: wsmaster/codenvy-hosted-api-organization/src/main/java/com/codenvy/organization/api/resource/SuborganizationResourcesProvider.java [hunk 3]
```diff
@@ -46,37 +55,67 @@
...
         return Collections.emptyList();
     }
+
+    private Optional<? extends Resource> findTimeoutResource(List<? extends Resource> resources) {
+        return resources.stream()
+                        .filter(resource -> resource.getType().equals(TimeoutResourceType.ID))
+                        .findAny();
+    }
+
+    private List<Resource> getDistributedResources(String accountId) throws ServerException {
```
Comment (evoevodin) [inline wsmaster/codenvy-hosted-api-organization/src/main/java/com/codenvy/organization/api/resource/SuborganizationResourcesProvider.java]: Does it makes sense to have such method in `OrganizationResourcesDistributor` ?

Comment (sleshchenko) [inline wsmaster/codenvy-hosted-api-organization/src/main/java/com/codenvy/organization/api/resource/SuborganizationResourcesProvider.java]: Good idea. In this way it looks much better

**Comment by sleshchenko** [comment]
Depends on https://github.com/eclipse/che/issues/4190

**Comment by sleshchenko** [comment]
Renaming of environment property is delayed because of back forwards compatibility problems. Will be created another PR for environment property renaming which will be merged after https://github.com/eclipse/che/issues/4190 closing.

**Comment by slemeur** [comment]
+1 ok for me.

**Comment by sleshchenko** [comment]
Created PR for renaming property https://github.com/codenvy/codenvy/pull/1838 when [issue](https://github.com/eclipse/che/issues/4190) with deprecating environment properties will be resolved.

**Pull request merged**